engine/execution/computation/computer/uploader/gcp_uploader.go

```diff
@@ -39,20 +40,33 @@ func NewGCPBucketUploader(ctx context.Context, bucketName string, log zerolog.Lo
 	}, nil
 }
 
-func (u *GCPBucketUploader) Upload(computationResult *execution.ComputationResult) error {
+func (u *GCPBucketUploader) Upload(computationResult *execution.ComputationResult) (err error) {
+	var errs *multierror.Error
 
 	objectName := GCPBlockDataObjectName(computationResult)
 	object := u.bucket.Object(objectName)
 
 	writer := object.NewWriter(u.ctx)
+
 	defer func() {
-		err := writer.Close()
-		if err != nil {
-			u.log.Warn().Err(err).Str("object_name", objectName).Msg("error while closing GCP object")
+		// flush and close the stream
+		// this occasionally fails with HTTP 50x errors due to flakiness on the network/GCP API
+		closeErr := writer.Close()
+		if closeErr != nil {
+			errs = multierror.Append(errs, fmt.Errorf("error while closing GCP object: %w", closeErr))
 		}
+
+		err = errs.ErrorOrNil()
 	}()
 
-	return WriteComputationResultsTo(computationResult, writer)
+	// serialize and write computation result to upload stream
+	writeErr := WriteComputationResultsTo(computationResult, writer)
+	if writeErr != nil {
+		errs = multierror.Append(errs, fmt.Errorf("error while writing computation result to GCP object: %w", writeErr))
+		return
```

Comment:
nit: this return can be removed.

Reply:
I added it for completeness. that way if this code is ever extended to do something after that conditional, there's less chance the implementor forgets to re-add the return.